dockstore-client/src/test/java/io/dockstore/client/cli/LaunchTestIT.java

```diff
@@ -1190,6 +1202,91 @@ public void cwlMissingIdParameters() {
     }
 
+
+    private void downloadTargetEntry(WorkflowClient workflowClient, String toolPath) {
+        try {
+            workflowClient.downloadTargetEntry("quay.io/collaboratory/dockstore-tool-linux-sort:2.0.0", ToolDescriptor.TypeEnum.WDL, true);
+        } catch (IOException e) {
+
+        }
+    }
+
+    @Test
+    public void entry2jsonNoVersion() {
+
+        WorkflowVersion aWorkflowVersion1 = new WorkflowVersion();
+        aWorkflowVersion1.setName("master");
+        aWorkflowVersion1.setValid(false);
+        Date earlierDate = new Date(100L);
+        aWorkflowVersion1.setLastModified(earlierDate);
+
+        List<WorkflowVersion> listWorkflowVersions = new ArrayList<WorkflowVersion>();
+        listWorkflowVersions.add(aWorkflowVersion1);
+
+        Workflow workflow = new Workflow();
+        workflow.setWorkflowVersions(listWorkflowVersions);
+        workflow.setLastModified(1);
+
+        WorkflowsApi api = mock(WorkflowsApi.class);
+        UsersApi usersApi = mock(UsersApi.class);
+        Client client = new Client();
+
+        doReturn(workflow).when(api).getPublishedWorkflowByPath(anyString(), eq(null));
+
+        WorkflowClient workflowClient = new WorkflowClient(api, usersApi, client, false);
+
+        exit.expectSystemExit();
+        exit.checkAssertionAfterwards(
+                () -> assertTrue("output should include error message",
+                        systemErrRule.getLog().contains("Cannot use workflow version 'master'")));
+
+        try {
+            workflowClient.downloadTargetEntry("quay.io/collaboratory/dockstore-tool-linux-sort", ToolDescriptor.TypeEnum.WDL, false);
+        } catch (IOException e) {
+
+        }
+    }
+
+    @Test
+    public void entry2jsonBadVersion() {
+
+        WorkflowVersion aWorkflowVersion1 = new WorkflowVersion();
+        aWorkflowVersion1.setName("1.0.0");
+        aWorkflowVersion1.setValid(false);
+        Date laterDate = new Date(1000L);
+        aWorkflowVersion1.setLastModified(laterDate);
+
+        List<WorkflowVersion> listWorkflowVersions = new ArrayList<WorkflowVersion>();
+        listWorkflowVersions.add(aWorkflowVersion1);
+
+        Workflow workflow = new Workflow();
+        workflow.setWorkflowVersions(listWorkflowVersions);
+        workflow.setLastModified(1);
+
+        WorkflowsApi api = mock(WorkflowsApi.class);
+        UsersApi usersApi = mock(UsersApi.class);
+        Client client = new Client();
+
+        doReturn(workflow).when(api).getPublishedWorkflowByPath(anyString(), eq(null));
+
+        WorkflowClient workflowClient = new WorkflowClient(api, usersApi, client, false);
+
+        exit.expectSystemExit();
+        exit.checkAssertionAfterwards(
+                () -> assertTrue("output should include error message",
+                        systemOutRule.getLog().contains("Could not locate workflow with version '2.0.0'")));
+        exit.checkAssertionAfterwards(
```

Comment:
Not suure whether the second assertion overwrites the first, can play it safe and just check both conditions at once.
Although looking below, I think only the first is expected? A bit confusing

Reply:
Yes both error messages are expected. I combined the assertions.

Reply:
Ah, I see. Thanks for the explanation